local-volume/provisioner/cmd/main.go

```diff
@@ -50,12 +64,26 @@ func main() {
 	node := getNode(client, nodeName)
 
 	glog.Info("Starting controller\n")
-	controller.StartLocalController(client, &common.UserConfig{
+	procTable := deleter.NewProcTable()
+	go controller.StartLocalController(client, procTable, &common.UserConfig{
 		Node:            node,
 		DiscoveryMap:    provisionerConfig.StorageClassConfig,
 		NodeLabelsForPV: provisionerConfig.NodeLabelsForPV,
 		UseAlphaAPI:     provisionerConfig.UseAlphaAPI,
 	})
+
+	glog.Infof("Starting metrics server at %s\n", optListenAddress)
+	prometheus.MustRegister([]prometheus.Collector{
+		metrics.PersistentVolumeProvisionTotal,
+		metrics.PersistentVolumeProvisionLatency,
+		metrics.APIServerRequestsTotal,
+		metrics.APIServerRequestsFailedTotal,
+		metrics.APIServerRequestsLantency,
+		metrics.PersistentVolumeCleanupFailedTotal,
+		collectors.NewProcTableCollector(procTable),
+	}...)
+	http.Handle(optMetricsPath, promhttp.Handler())
+	log.Fatal(http.ListenAndServe(optListenAddress, nil))
```

Comment:
Have you had any issues shutting down the container with this new code? Does the daemonset stop responsively? or does the kubelet sleep for 30 seconds before doing a hard kill of the container? Just want to make sure this listen still responds to SIGINT.  If you do find that its not exiting quickly, you might need to add code along these lines: https://medium.com/@kpbird/golang-gracefully-stop-application-23c2390bb212

Reply:
Doesn't have any issues shutdown down the container with this new code. Container exits quickly after pod is deleted.